src/io/flutter/view/FlutterView.java

```diff
@@ -445,16 +445,20 @@ private void presentDevTools(FlutterApp app, InspectorService inspectorService,
   }
 
   protected void openInspectorWithDevTools(FlutterApp app, InspectorService inspectorService, ToolWindow toolWindow, boolean isEmbedded) {
-    ApplicationManager.getApplication().executeOnPooledThread(() -> {
-      final DevToolsInstance instance = DevToolsService.getInstance(myProject).getDevToolsInstance();
+    AsyncUtils.whenCompleteUiThread(DevToolsService.getInstance(myProject).devToolsInstance, (instance, error) -> {
+      // TODO(helinx): Restart DevTools server if there's an error.
+      if (error != null) {
```

Comment:
make sure the project has not gone away while you were waiting for the devToolsServer to start.